zScanner/ViewControllers/Documents/DocumentsListViewController.swift

```diff
@@ -72,62 +74,125 @@ class DocumentsListViewController: BaseViewController, ErrorHandling {
+                            DepartmentButton(model: $0)
+                        })
+                        .forEach({ (button) in
+                            button.rx.tap
+                                .subscribe({ [weak self] _ in
+                                    self?.loadDocumentTypes(departmentCode: button.model.id)
+                                })
+                                .disposed(by: self.disposeBag)
+
+                            self.departmentsStackView.addArrangedSubview(button)
+                        })
+                    
+                case .error( _):
+                    self.departmentsStackView.subviews.forEach({ $0.removeFromSuperview() })
+                    
+                    self.departmentsStackView.addArrangedSubview(self.departmentsErrorView)
+                    
+                }
+            })
+            .disposed(by: disposeBag)
+        
+        departmentsErrorView.buttonTap
+            .subscribe(onNext: { [weak self] _ in
+                self?.departmentsViewModel.fetchDepartments()
+            })
+        .disposed(by: disposeBag)
     }
     
     @objc private func openMenu() {
         coordinator.openMenu()
     }
-    
-    @objc private func reloadDocumentTypes() {
-        viewModel.updateDocumentTypes()
+
+    private func loadDocumentTypes(departmentCode: String) {
+        documentsViewModel.fetchDocumentTypes(for: departmentCode)
     }
     
     private func setupView() {
         navigationItem.title = "document.screen.title".localized
         
-        view.addSubview(tableView)
-        tableView.snp.makeConstraints { make in
-            make.edges.equalToSuperview()
+        documentsTableView.dataSource = self
+        view.addSubview(documentsTableView)
+        documentsTableView.snp.makeConstraints { (make) in
+            make.top.trailing.leading.equalToSuperview()
         }
         
-        tableView.backgroundView = emptyView
+        documentsTableView.backgroundView = emptyView
         
         emptyView.addSubview(emptyViewLabel)
         emptyViewLabel.snp.makeConstraints { make in
             make.width.equalToSuperview().multipliedBy(0.75)
             make.centerX.equalToSuperview()
-            make.top.greaterThanOrEqualTo(tableView.safeAreaLayoutGuide.snp.top)
+            make.top.equalTo(documentsTableView.sectionHeaderHeight)
             make.centerY.equalToSuperview().multipliedBy(0.666).priority(900)
         }
+        
+        view.addSubview(departmentsContainerView)
+        departmentsContainerView.snp.makeConstraints { (make) in
+            make.top.equalTo(documentsTableView.snp.bottom).offset(16)
+            make.left.right.bottom.equalToSuperview()
+            make.height.equalTo(0).priority(100)
+            make.height.lessThanOrEqualTo(view.snp.height).multipliedBy(0.666)
+        }
+        
+        departmentsContainerView.backgroundColor = UIColor.lightGray.withAlphaComponent(0.3)
```

Comment:
Please create lazy init for `departmentsContainerView` and move the background colour there.  I did it yesterday in a rush.